lib/ansible/plugins/become/runas.py

```diff
@@ -0,0 +1,74 @@
+# Copyright: (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    become: runas
+    short_description: Run As user
+    description:
+        - This become plugins allows your remote/login user to execute commands as another user via the windows runas facility.
+    author: ansible (@core)
+    version_added: "2.8"
+    options:
+        become_user:
+            description: User you 'become' to execute the task
+            ini:
+              - section: privilege_escalation
+                key: become_user
+              - section: runas_become_plugin
+                key: user
+            vars:
+              - name: ansible_become_user
+              - name: ansible_runas_user
+            env:
+              - name: ANSIBLE_BECOME_USER
+              - name: ANSIBLE_RUNAS_USER
+            required: True
+        become_exe:
+            description: Sudo executable
+            default: runas
+            ini:
+              - section: privilege_escalation
+                key: become_exe
+              - section: runas_become_plugin
+                key: executable
+            vars:
+              - name: ansible_become_exe
+              - name: ansible_runas_exe
+            env:
+              - name: ANSIBLE_BECOME_EXE
+              - name: ANSIBLE_RUNAS_EXE
+        become_flags:
+            description: Options to pass to runas
+            default: -H -S -n
+            ini:
+              - section: privilege_escalation
+                key: become_flags
+              - section: runas_become_plugin
+                key: flags
+            vars:
+              - name: ansible_become_flags
+              - name: ansible_runas_flags
+            env:
+              - name: ANSIBLE_BECOME_FLAGS
+              - name: ANSIBLE_RUNAS_FLAGS
+    notes:
+        - runas is really implemented in the winrm conneciton plugin and as such can only be used with winrm connections.
```

Comment:
It's more implemented in the powershell shell plugin than winrm. We have both psrp and winrm that support runas as they are both Windows specific. When finally adding support for SSH connections to Windows then it would also be supported there.